lib/util/determine-files.js

```diff
@@ -0,0 +1,81 @@
+let path = require("path");
+let fs = require("fs");
+
+/* Returns a function that determines files to process
+ *
+ * This is useful for pipelines that process a directory of files.
+ * It takes a source directory and optionally a predicate that is used to filter
```

Comment:
`predicate` seems like an unnecessarily pretentious term; it's just a function, yes?